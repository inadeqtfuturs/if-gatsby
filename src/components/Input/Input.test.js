import React from 'react';
import { Input } from '@components';
import ThemeRender from '@utils/testing';

describe('Input', () => {
  it('renders correctly', () => {
    const tree = ThemeRender(<Input name="test" />);
    expect(tree).toMatchSnapshot();
  });

  it('takes props', () => {
    const tree = ThemeRender(
      <Input
        label="name"
        name="name"
        placeholder="full name"
        size="lg"
        disabled
        required
        row
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('displays help text', () => {
    const tree = ThemeRender(
      <Input label="help" name="help" helpText="here's some help text" />
    );
    expect(tree).toMatchSnapshot();
  });

  it('displays error state', () => {
    const tree = ThemeRender(
      <Input
        label="error"
        name="error"
        meta={{ touched: true, submitError: true, error: true }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
