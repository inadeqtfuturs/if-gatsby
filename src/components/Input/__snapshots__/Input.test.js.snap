// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input displays error state 1`] = `
<div
  className="sc-fzoLsD kqvQGr"
>
  <label
    className="sc-fznyAO KFPww css-knfhfe"
    htmlFor="error"
    required={false}
    size="md"
  >
    error
  </label>
  <input
    className="sc-fzplWN eYgDYr css-iyqgnx"
    disabled={false}
    name="error"
    placeholder={null}
    size="md"
  />
</div>
`;

exports[`Input displays help text 1`] = `
<div
  className="sc-fzoLsD kqvQGr"
>
  <label
    className="sc-fznyAO KFPww css-knfhfe"
    htmlFor="help"
    required={false}
    size="md"
  >
    help
  </label>
  <input
    className="sc-fzplWN AecOf css-iyqgnx"
    disabled={false}
    name="help"
    placeholder={null}
    size="md"
  />
  <small
    className="sc-fzpans kkFAbe"
  >
    here's some help text
  </small>
</div>
`;

exports[`Input renders correctly 1`] = `
<div
  className="sc-fzoLsD kqvQGr"
>
  <input
    className="sc-fzplWN AecOf css-iyqgnx"
    disabled={false}
    name="test"
    placeholder={null}
    size="md"
  />
</div>
`;

exports[`Input takes props 1`] = `
<div
  className="sc-fzoLsD dDqpSf"
>
  <label
    className="sc-fznyAO VGARM css-knfhfe"
    htmlFor="name"
    required={true}
    size="lg"
  >
    name
  </label>
  <input
    className="sc-fzplWN eJwMql css-iyqgnx"
    disabled={true}
    name="name"
    placeholder="full name"
    size="lg"
  />
</div>
`;
